sqlgen/reflect.go

```diff
@@ -243,15 +243,14 @@ func (s *Schema) buildDescriptor(table string, primaryKeyType PrimaryKeyType, ty
 			return nil, fmt.Errorf("bad type %s: duplicate column %s", typ, column)
 		}
 
-		var scannable func() Scannable
-
+		scalarType := field.Type
 		if field.Type.Kind() == reflect.Ptr {
-			scannable, _ = s.scalarTypes[field.Type.Elem()]
-		} else {
-			scannable, _ = s.scalarTypes[field.Type]
+			scalarType = field.Type.Elem()
 		}
 
-		if scannable == nil {
+		scannable, ok := s.scalarTypes[scalarType]
+		scaninterface := reflect.TypeOf((*Scannable)(nil)).Elem()
+		if !ok && !scalarType.Implements(scaninterface) && !reflect.PtrTo(scaninterface).Implements(scaninterface) {
```

Comment:
We have to check both the element and its pointer in case one of those satisfies the interface. This is because `(t *T) Scan()` on non-pointer `T` won't register as implementing the interface (despite being able to call that method on it)